### PR TITLE
Draco fix for #6523

### LINKED ISSUE #6523: Model shader error for Draco + morph targets
[cube-draco-morph-targets.gltf.txt](https://github.com/AnalyticalGraphicsInc/cesium/files/1960731/cube-draco-morph-targets.gltf.txt)

error

I was trying out the model from [here](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/issues/365) but it fails during shader compilation.

@ggetz would you be able to look into this?

### PR DESCRIPTION
**Pull request by ggetz**
Fixes #6523

`replaceAllButFirstInString` was replacing the variable name everywhere, even in subsets of other variables, (ex. `a_position` was replaced in both `a_position` and `a_position_1`).

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@ggetz, thanks for the pull request! Maintainers, we have a signed CLA from @ggetz, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Source/Scene/ModelUtility.js [hunk 1]
```diff
@@ -167,9 +167,18 @@ define([
...
+    function regexIndexOf(text, regex, i) {
+        var indexInSuffix = text.slice(i).search(regex);
+        return indexInSuffix < 0 ? indexInSuffix : indexInSuffix + i;
+    }
+
     function replaceAllButFirstInString(string, find, replace) {
-        var index = string.indexOf(find);
-        return string.replace(new RegExp(find, 'g'), function(match, offset) {
+        // Limit search to strings that are not a subset of other tokens.
+        find += '(?![a-zA-Z_$])';
```
Comment (lilleyse) [inline Source/Scene/ModelUtility.js]: Can the `$` be removed?

File: Source/Scene/ModelUtility.js [hunk 2]
```diff
@@ -167,9 +167,18 @@ define([
...
+        find = new RegExp(find, 'g');
+
+        var index = regexIndexOf(string, find, 0);
```
Comment (lilleyse) [inline Source/Scene/ModelUtility.js]: Could this line just be replaced with `var index = string.search(find);` and `regexIndexOf` be removed?

File: Source/Scene/ModelUtility.js (hunk 1 above)
Comment (ggetz) [reply Source/Scene/ModelUtility.js]: Yes, also realized I missed `0-9`.

Comment (lilleyse) [reply Source/Scene/ModelUtility.js]: I guess in that case, could `\w` be used instead?

Comment (ggetz) [reply Source/Scene/ModelUtility.js]: Yep

**Comment by lilleyse** [comment]
Thanks!

**Pull request merged**